x-pack/filebeat/input/netflow/case.go

```diff
@@ -0,0 +1,87 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package netflow
+
+import (
+	"strings"
+	"sync"
+	"unicode"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/x-pack/filebeat/input/netflow/decoder/record"
+)
+
+var fieldNameConverter = caseConverter{
+	conversion: make(map[string]string),
+}
+
+type caseConverter struct {
+	rwMutex    sync.RWMutex
+	conversion map[string]string
+}
+
+func (c *caseConverter) memoize(nfName, converted string) string {
+	c.rwMutex.Lock()
+	defer c.rwMutex.Unlock()
+	c.conversion[nfName] = converted
+	return converted
+}
+
+func (c *caseConverter) ToSnakeCase(orig record.Map) common.MapStr {
+	result := common.MapStr(make(map[string]interface{}, len(orig)))
+	c.rwMutex.RLock()
+	defer c.rwMutex.RUnlock()
+
+	for nfName, value := range orig {
+		name, found := c.conversion[nfName]
+		if !found {
+			c.rwMutex.RUnlock()
+			name = c.memoize(nfName, CamelCaseToSnakeCase(nfName))
+			c.rwMutex.RLock()
+		}
+		result[name] = value
+	}
+	return result
+}
+
+// CamelCaseToSnakeCase converts a camel-case identifier to snake-case
+// format. This function is tailored to some specifics of NetFlow field names.
+// Don't reuse it.
+func CamelCaseToSnakeCase(in string) string {
```

Comment:
well, it will be used as netflow.CamelCaseToSnakeCase when imported from outside.

It's a public (exported) method because the field generator (fields_gen.go) must be in main package and use this method.